src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -147,14 +147,19 @@ public static JsonDocument Parse(string json, JsonReaderOptions readerOptions =
             JsonReaderOptions readerOptions,
             byte[] extraRentedBytes)
         {
+            if (readerOptions.MaxDepth == 0)
+            {
+                readerOptions.MaxDepth = JsonReaderOptions.DefaultMaxDepth;
```

Comment:
Isn't this redundant with the reader's version of this?

Reply:
Yep, removed.